R/data-storage-mssqlserver.R

```diff
@@ -0,0 +1,126 @@
+#'
+#' @description
+#' Implementation of the DataStorage R6 class to MS SQL Server backend using a
+#' unified API for read/write operations
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' data_storage <- DataStorageMSSQLServer$new(user = "sqlserver", 
+#' password = "mysecretpassword", hostname = "servername", port = 1433,
+#' dbname = "my_db", driver = "ODBC Driver 17 for SQL Server", 
+#' trust_server_certificate = "NO")
+#'
+#' data_storage$insert("example", "test_event", "session1")
+#' data_storage$insert("example", "input", "s1", list(id = "id1"))
+#' data_storage$insert("example", "input", "s1", list(id = "id2", value = 32))
+#'
+#' data_storage$insert(
+#'   "example", "test_event_3_days_ago", "session1",
+#'   time = lubridate::as_datetime(lubridate::today() - 3)
+#' )
+#'
+#' data_storage$read_event_data()
+#' data_storage$read_event_data(Sys.Date() - 1, Sys.Date() + 1)
+#' data_storage$close()
+#' }
+DataStorageMSSQLServer <- R6::R6Class( # nolint object_name_linter
+  classname = "DataStorageMSSQLServer",
+  inherit = DataStorageSQLFamily,
+  #
+  # Public
+  public = list(
+
+    #' @description
+    #' Initialize the data storage class
+    #' @param username string with a MS SQL Server username.
+    #' @param password string with the password for the username.
+    #' @param hostname string with hostname of the MS SQL Server instance.
+    #' @param port numeric value with the port number of MS SQL Server instance.
+    #' @param dbname string with the name of the database in the MS SQL Server
+    #' @param driver string with the name of the ODBC driver class for MS SQL.
+    #' @param trust_server_certificate string with "NO" or "YES", setting 
+    #' whether or not to trust the server's certificate implicitly.
+    #' instance.
+
+    initialize = function(
+      username = NULL,
+      password = NULL,
+      hostname = "127.0.0.1",
+      port = 1433,
+      dbname = "shiny_telemetry",
+      driver = "ODBC Driver 17 for SQL Server",
+      trust_server_certificate = "NO"
+    ) {
+      checkmate::assert_string(password)
+      checkmate::assert_string(username)
+      checkmate::assert_string(hostname)
+      checkmate::assert_int(port)
+      checkmate::assert_string(dbname)
+      checkmate::assert_string(driver)
+      checkmate::assert_string(trust_server_certificate)
+
+      logger::log_debug(
+        "Parameters for MS SQL Server:\n",
+        "  *                 username: {username}\n",
+        "  *        password (sha256): {digest::digest(password, algo = 'sha256')}\n",
+        "  *            hostname:port: {hostname}:{port}\n",
+        "  *                  db name: {dbname}\n",
+        "  *                   driver: {driver}\n",
+        "  * trust_server_certificate: {trust_server_certificate}\n",
+        namespace = "shiny.telemetry"
+      )
+      private$connect(username, password, hostname, port, dbname, driver,
+                      trust_server_certificate)
```

Comment:
Can you make this small change to follow the styling of the rest of the package, i.e., prefer a newline when function arguments span multiple lines.

Thanks!

```suggestion
      private$connect(
        username,
        password,
        hostname,
        port,
        dbname,
        driver,
        trust_server_certificate
      )
```

Reply:
Hi @averissimo - sure - done in latest commit :)